services/device-registry-base/src/main/java/org/eclipse/hono/service/management/tenant/TrustedCertificateAuthority.java

```diff
@@ -374,4 +419,98 @@ public final TrustedCertificateAuthority setAuthIdTemplate(final String template
         return this;
     }
 
+    /**
+     * Gets whether OCSP revocation check is enabled. It is disabled by default.
+     *
+     * @return True if OCSP revocation is enabled.
+     */
+    public boolean isOcspRevocationEnabled() {
+        return ocspRevocationEnabled;
+    }
+
+    /**
+     * Sets whether OCSP revocation check is enabled. It is disabled by default. If OCSP is enabled then trusted CA
+     * certificate or OCSP responder certificate must be explicitly set.
+     *
+     * @param ocspRevocationEnabled True if OCSP revocation check should be enabled.
+     * @return A reference to this for fluent use.
+     */
+    public TrustedCertificateAuthority setOcspRevocationEnabled(final boolean ocspRevocationEnabled) {
+        this.ocspRevocationEnabled = ocspRevocationEnabled;
+        return this;
+    }
+
+    /**
+     * Gets OCSP responder uri which will be used for OCSP revocation check of this trust anchor. If URI is null then
+     * certificate AIA extension value should be used to obtain the values instead.
+     *
+     * @return OCSP responder URI or null.
+     */
+    public String getOcspResponderUri() {
+        return ocspResponderUri;
+    }
+
+    /**
+     * Sets OCSP responder uri which will be used for OCSP revocation check of this trust anchor. If URI is null then
+     * certificate AIA extension value should be used to obtain the values instead.
+     *
+     * @param ocspResponderUri OCSP responder URI or null to use certificate AIA extension instead.
+     * @return A reference to this for fluent use.
+     */
+    public TrustedCertificateAuthority setOcspResponderUri(final String ocspResponderUri) {
+        this.ocspResponderUri = ocspResponderUri;
+        return this;
+    }
+
+    /**
+     * Gets OCSP responder certificate which is used to verify OCSP response signature. If custom certificate is not
+     * set then issuing CA of client certificate is used instead.
```

Comment:
IMHO this will only work, if the user also includes the trust anchor certificate in its request, because otherwise *cert* will always be `null` as we do not store the cert in the DB but only its public key and subject DN.
This means that we *can not* later enable OCSP using the trust anchor CA for this purpose on an already existing trust anchor, or am I mistaken?

Reply:
Technically yes. Since OCSP implementation requires full certificate, user must provide it either in cert or [ocsp-responder-cert](https://github.com/kyberpunk/hono/blob/enable_revocation_check_for_trusted_ca/site/documentation/content/api/management/device-registry-v1.yaml#L944) property. I may test what happens in such case, when you just enable it by PUT, but it should be handled by [validation](https://github.com/kyberpunk/hono/blob/enable_revocation_check_for_trusted_ca/services/device-registry-base/src/main/java/org/eclipse/hono/service/management/tenant/TrustedCertificateAuthority.java#L108).